### DISCUSSION TITLE
Python version support policy

### DISCUSSION THREAD

**Discussion by peterschutt** [litestar-org/litestar#713] [Announcements]
Maintainers have been discussing creating a policy for python version support, and are considering making the policy span 4 current versions.

This will mean we drop python 3.7 support before its official EOL.

Anyone out there using starlite on 3.7 that would be adversely affected by this change?

**Comment by Goldziher** [comment]
i suggest that if we don't see a blocker for this emerging in the next 2 weeks, we make this actionable.

**Comment by jonadaly** [comment]
For posterity, why are we thinking of doing this?

I can think of a few advantages of dropping 3.7 support before its EOL in June 2023:
- Slightly simpler CI
- Syntax for 3.8 without `future` (does this affect us?)
- Support for >3.8 via `future` (e.g. generic types)

Disadvantages:
- Anyone on 3.7 using starlite won't receive any new features/bugfixes (I imagine this is a vanishingly small % of users)
- If we're aiming to be a widely used framework, perhaps we ought to support all active versions of python

**Reply by Goldziher** [reply]
Biggest difference is the `Literal` type and `TypedDict `

**Reply by jtraub** [reply]
But both are available through `typing_extensions` 🤔

**Comment by Goldziher** [comment]
ok, we are removing 3.7
